### PR TITLE
gpu enabled docker file with python bindings

### PR DESCRIPTION
**Pull request by nji-syd**

### CODE REVIEW

**Comment by guolinke** [comment]
@wxchan

**Comment by wxchan** [comment]
@guolinke I think @huanzhang12 has reviewed this.

**Comment by guolinke** [comment]
@nji-syd can you sigh the cla ?

File: docker/gpu/README.md [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+jupyter password: keras
+### Requirements
+Requires docker and [nvidia-docker](https://github.com/NVIDIA/nvidia-docker) on host machine.
+### Quickstart
+
+##### Build Docker Image
+```sh
+mkdir lightgbm-docker
+cd lightgbm-docker
+git clone --recursive https://github.com/nji-syd/lightgbm-docker
```
Comment (StrikerRUS) [inline docker/gpu/README.md]: I think after merging it should be something like this:
```
mkdir lightgbm-docker
cd lightgbm-docker
wget https://github.com/Microsoft/LightGBM/blob/master/docker/gpu/dockerfile.gpu
docker build -f dockerfile.gpu -t lightgbm-gpu .
```

Comment (huanzhang12) [reply docker/gpu/README.md]: Yes the README.md should be updated with link to LightGBM. I agree with the commands @StrikerRUS proposed.

**Comment by huanzhang12** [comment]
I reviewed and tested the Dockerfile and it works great. This will greatly help users with GPU setup. Thanks @nji-syd for your contribution.

**Comment by nji-syd** [comment]
base image nvidia/cuda:8.0-cudnn5-devel is 1.8GB while nvidia/cuda:8.0-cudnn5-runtime-ubuntu16.04 is 0.9 GB, however the runtime is missing some required packages. It can probably be made leaner. thanks for the troubleshooting @huanzhang12

File: docker/gpu/README.md (hunk 1 above)
Comment (guolinke) [reply docker/gpu/README.md]: @nji-syd could you fix this ?

**Comment by nji-syd** [comment]
@guolinke @StrikerRUS change has been pushed

**Pull request merged**